realm/src/main/java/io/realm/internal/InvalidRow.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.realm.internal;
+
+import java.util.Date;
+
+/**
+ * Row wrapper that stubs all access with IllegalStateExceptions. This can be used instead of adding null checks
+ * everywhere when the underlying Row accessor in Realm Core is no longer available.
+ */
+public class InvalidRow implements Row {
```

Comment:
Should be final (to avoid accidentally extending) & Probably singleton since this is immutable (no need to create a new instance for each InvalidRow).